### PR TITLE
add link to parent of comment

### PR DESCRIPTION
**Pull request by xundeenergie**
In longer disussions/threads it is almost impossible to know, to whom/which comment a comment belongs.

This PR adds a link to the parent item, where a comment belongs to.

grafik

It opens the comment/posting with "display/<guid>" and adds the original position to history, so you can go back with the back-button of your browser to where you started.

### CODE REVIEW

**Comment by xundeenergie** [comment]
a little improvement

On wide-screen
grafik

On wide-screen, when comment is reshared
grafik

The same on xs-screen
grafik

On xs-screen when reshared
grafik

There is a dot between answer-to and reshared-by on wide-screen
and a `<br>` on xs-screen instead.

File: view/theme/frio/templates/wall_thread.tpl [hunk 1]
```diff
@@ -59,9 +59,15 @@ as the value of $top_child_total (this is done at the end of this file)
 <span class="uriid" style="display: none;">{{$item.uriid}}</span>
 {{/if}}
 	<div class="media {{$item.shiny}}">
+	{{if !$item.parentguid==""}}
```
Comment (annando) [inline view/theme/frio/templates/wall_thread.tpl]: Doesn't `{{if !$item.parentguid}}` work as well?

Comment (xundeenergie) [reply view/theme/frio/templates/wall_thread.tpl]: I didn't try. Should I?

Comment (MrPetovan) [reply view/theme/frio/templates/wall_thread.tpl]: It should be 
```suggestion
	{{if $item.parentguid}}
```

File: src/Object/Post.php [hunk 2]
```diff
@@ -497,6 +497,8 @@ public function getTemplateData(array $conv_responses, string $formSecurityToken
 		}
 
 		$tmp_item = [
+			'parentguid'      => $parent_guid,
+			'isanswerto'      => DI::l10n()->t('is answer to %s', $parent_username),
```
Comment (MrPetovan) [inline src/Object/Post.php]:
```suggestion
			'isanswerto'      => DI::l10n()->t('is reply to %s', $parent_username),
```

File: view/theme/frio/js/theme.js [hunk 3]
```diff
@@ -184,6 +184,7 @@ $(document).ready(function () {
 		if (typeof searchValue !== "undefined") {
 			$("#nav-search-input-field").val(searchValue);
 		}
+
```
Comment (MrPetovan) [inline view/theme/frio/js/theme.js]: Please remove this empty line.
```suggestion
```

File: view/theme/frio/templates/wall_thread.tpl (hunk 1 above)
Comment (xundeenergie) [reply view/theme/frio/templates/wall_thread.tpl]: seems it works

File: view/theme/frio/js/theme.js [hunk 4]
```diff
@@ -941,4 +942,13 @@ function toggleDropdownText(elm) {
...
 }
+
+// Go to parent for "is answer to <user>" link in comments below username
+function goToParentItem(plink, pguid) {
+	history.pushState({},"",plink);
+	var loc = plink.split("/");
+	loc.pop();
+	loc.push(pguid);
+	location.href = loc.join("/");
+}
```
Comment (MrPetovan) [inline view/theme/frio/js/theme.js]: There's already a `scrollToItem()` function in the same file that you can use for this.

Here's the usage: `scrollToItem("item-" + itemGuid)`.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: view/theme/frio/js/theme.js (hunk 4 above)
Comment (xundeenergie) [reply view/theme/frio/js/theme.js]: oh wonderful. i was looking for exactly this. :)

File: view/theme/frio/js/theme.js (hunk 3 above)
Comment (xundeenergie) [reply view/theme/frio/js/theme.js]: done

File: src/Object/Post.php (hunk 2 above)
Comment (xundeenergie) [reply src/Object/Post.php]: renamed the variable too in all places.

**Comment by xundeenergie** [comment]
How should i handle such conflicts in messages.po?

**Comment by MrPetovan** [comment]
Just run the `bin/run_xgettext.sh` script again and commit the changes.

File: src/Object/Post.php [hunk 5]
```diff
@@ -497,6 +497,8 @@ public function getTemplateData(array $conv_responses, string $formSecurityToken
 		}
 
 		$tmp_item = [
+			'parentguid'      => $parent_guid,
+			'isreplyto'      => DI::l10n()->t('is reply to %s', $parent_username),
```
Comment (MrPetovan) [inline src/Object/Post.php]: Missing space to align the arrows.

Also no need to use `DI` in this case, the translation methods are available to all classes inheriting from `BaseModule`.
```suggestion
			'isreplyto'       => $this->t('is reply to %s', $parent_username),
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Object/Post.php (hunk 5 above)
Comment (xundeenergie) [reply src/Object/Post.php]: should i change the other usages of DI in this place too?

Comment (xundeenergie) [reply src/Object/Post.php]: Sorry... no it does not work. When i make your changes, i get a blank white page.

In friendica.log this:
```
2023-10-06T16:57:12Z app [ERROR]: Uncaught Exception Error: "Call to undefined method Friendica\Object\Post::t()" at /var/www/soc.schuerz.at/src/Object/Post.php line 501 {"exception":"Error: Call to undefined method Friendica\\Object\\Post::t() in /var/www/soc.schuerz.at/src/Object/Post.php:501\nStack trace:\n#0 /var/www/soc.schuerz.at/src/Object/Thread.php(204): Friendica\\Object\\Post->getTemplateData()\n#1 /var/www/soc.schuerz.at/src/Content/Conversation.php(685): Friendica\\Object\\Thread->getTemplateData()\n#2 /var/www/soc.schuerz.at/src/Content/Conversation.php(583): Friendica\\Content\\Conversation->getThreadList()\n#3 /var/www/soc.schuerz.at/src/Module/Item/Display.php(278): Friendica\\Content\\Conversation->render()\n#4 /var/www/soc.schuerz.at/src/Module/Item/Display.php(165): Friendica\\Module\\Item\\Display->getDisplayData()\n#5 /var/www/soc.schuerz.at/src/BaseModule.php(246): Friendica\\Module\\Item\\Display->content()\n#6 /var/www/soc.schuerz.at/src/App.php(710): Friendica\\BaseModule->run()\n#7 /var/www/soc.schuerz.at/index.php(46): Friendica\\App->runFrontend()\n#8 {main}"} - {"file":null,"line":null,"function":null,"request-id":"65203c685ec37","uid":"78691a","process_id":3269014}
```

**Comment by xundeenergie** [comment]
> Just run the `bin/run_xgettext.sh` script again and commit the changes.

I made a misstage... have a split brain between the local branch "reference-links" and the one on github.

image

The commits have the same comments, but they are different.
`211fbce9c translations update`
is not the same as 
`428884c62 translations update`

i think i made the misstake on rebasing and pushing... 

And i have no idea, how to fix this.

My changes are in both branches (the local and the remote one). 
The only idea i have is to delete the remote branch and push it again. But this will break this pull-request... 

How can i solve this situation?

File: src/Object/Post.php (hunk 5 above)
Comment (MrPetovan) [reply src/Object/Post.php]: Sorry about that, the `Object\Post` class doesn't inherit from BaseModule and as such doesn't have the `t()` method available.

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**